tests/util.test.ts

```diff
@@ -0,0 +1,167 @@
+import { assert, create, is, cast, type, valid, validate } from '../util'
+
+describe('create', () => {
+  it('works with different types', () => {
+    expect(create('Thing')).toEqual({
+      type: 'Thing'
+    })
+    expect(create('CreativeWork')).toEqual({
+      type: 'CreativeWork'
+    })
+  })
+
+  it('works with initial values', () => {
+    expect(
+      create('Thing', {
+        name: 'thing1',
+        url: 'http://example.com/thing1'
+      })
+    ).toEqual({
+      type: 'Thing',
+      name: 'thing1',
+      url: 'http://example.com/thing1'
+    })
+  })
+
+  it('throws with unknown types', () => {
+    // In Typescript this error is caught at compile time, so ts-ignore it
+    // @ts-ignore
+    expect(() => create('Foo')).toThrow(/^No schema for type "Foo".$/)
+  })
+
+  it('throws when wrong initial values', () => {
+    expect(() => create('Thing', { foo: 'Foo' })).toThrow(
+      'Property foo is not expected to be here'
+    )
+    expect(() => create('Thing', { type: 'Foo' })).toThrow(
+      'type should be equal to one of the allowed values: Thing'
+    )
+  })
+
+  it('does not throw when wrong initial values and no validation', () => {
+    expect(create('Thing', { foo: 'invalid' }, false)).toEqual({
+      type: 'Thing',
+      foo: 'invalid'
+    })
+  })
+})
+
+test('type', () => {
+  expect(type(undefined)).toBe('undefined')
+  expect(type(null)).toBe('null')
+  expect(type(true)).toBe('boolean')
+  expect(type(0)).toBe('number')
```

Comment:
Ah yes, I think the only type missing is:
``` ts
expect(type(NaN)).toBe('number')
```

Reply:
:+1: